doc/HOWTO/documenting-functions-and-macros.md

```diff
@@ -0,0 +1,175 @@
+Documenting public Functions and Macros
+=======================================
+
+In the last years, the OpenSSL project has strived to improve the quality
+and coverage of the API documentation. A while ago, this goal has been
+turned into an official [documentation-policy]. This policy is actively
+enforced by the `make doc-nits` target resp. `check-docs` GitHub action.
+
+[documentation-policy]: https://www.openssl.org/policies/technical/documentation-policy.html
+
+If you add a new public function or macro to a header file without documenting
+it, it will give you an error message like this:
+
+```text
+include/openssl/bio.h: macro BIO_set_dgram_origin(3) undocumented
+include/openssl/bio.h: macro BIO_get_dgram_origin(3) undocumented
+include/openssl/bio.h: macro BIO_set_dgram_dest(3) undocumented
+include/openssl/bio.h: macro BIO_get_dgram_dest(3) undocumented
+```
+
+and you'll want to document this.
+
+So, create a new .pod file named doc/man3/FUNCTION.pod.
+
+If you are asked to document several related functions in that file,
+you can create a single pod file in which you document them together.
+In this case, use the name of the first function as the file name,
+like for the above example:
+
+```text
+doc/man3/BIO_set_dgram_origin.pod
+```
+
+If you do use an unrelated name (like BIO_dgram.pod) then you'll get
+a warning about that.
+
+Next, you need to add your new file to the `doc/build.info` file.
+This command does it automatically for you:
+
+```console
+$ make generate_doc_buildinfo
+```
+
+this will update doc/build.info.
+You should git add the result as `generate_doc_buildinfo` is not run on every build.
+
+With these two changes, running `make doc-nits` locally should
+now agree with you that you have documented all your new defines,
+but it might then complain:
+
+```text
+BIO_get_dgram_dest(3) is supposedly internal
+(maybe missing from other.syms) but is documented as public
+```
+
+If it is the case that your interface is meant to be public, then you need
+to edit the file `util/other.syms` to add the names of your `#define`
+functions.
+This file gets sorted alphabetically prior to each major release,
+but new additions should be placed at the end of the file.
+
+### Example
+
+For demonstration purposes, two new public symbols have been added
+by "implementing" a public function `BIO_set_dgram_foo()`
+and a public function-like macro `BIO_set_dgram_bar()`:
+
+```diff
+diff --git a/crypto/bio/bss_dgram.c b/crypto/bio/bss_dgram.c
+index 82d382cf4e..30382f0abe 100644
+--- a/crypto/bio/bss_dgram.c
++++ b/crypto/bio/bss_dgram.c
+@@ -192,6 +192,13 @@ BIO *BIO_new_dgram(int fd, int close_flag)
+     return ret;
+ }
+
++
++int BIO_set_dgram_foo(BIO* b, int foo)
++{
++    return foo;
++}
++
++
+ static int dgram_new(BIO *bi)
+ {
+     bio_dgram_data *data = OPENSSL_zalloc(sizeof(*data));
+diff --git a/include/openssl/bio.h.in b/include/openssl/bio.h.in
```

Comment:
Likely due to a bug in the GitHub markdown handling, the second half of this quoted diff is not as colorful as the first one.
I suppose a workaround would be to finish the first one and start a new diff section here, using again
```
```diff
```

Reply:
It's not a bug in the GitHub markdown handling, it's a bug in copy&pasting 😉  : the bottom hunk has leading spaces.

I'd suggest that you just recopy the [original patch](https://github.com/openssl/openssl/pull/17926#issuecomment-1082407976) which is coloured correctly.